step/step.go

```diff
@@ -0,0 +1,457 @@
+package step
```

Comment:
Nit: I'd structure the file in this order:
```
package step

import (
    ...
)

<struct definitions here>

<other custom types here>

<const definitions here>

<file-level var definitions here>

<functions here (preferably exported first)>
```

It's not a big deal, but it would be a bit clearer I think.

Reply:
I agree with the structure, done in latest commit